hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigurationAwareConfig.java

```diff
@@ -1113,16 +973,24 @@ public EventJournalConfig getCacheEventJournalConfig(String name) {
     }
 
     private EventJournalConfig getCacheEventJournalConfigInternal(String name, String fallbackName) {
-        String baseName = getBaseName(name);
-        Map<String, EventJournalConfig> cacheEventJournalConfigs = staticConfig.getCacheEventJournalConfigs();
-        EventJournalConfig eventJournalConfig = lookupByPattern(configPatternMatcher, cacheEventJournalConfigs, baseName);
-        if (eventJournalConfig == null) {
-            eventJournalConfig = configurationService.findCacheEventJournalConfig(baseName);
-        }
-        if (eventJournalConfig == null) {
-            eventJournalConfig = staticConfig.getCacheEventJournalConfig(fallbackName);
-        }
-        return eventJournalConfig;
+        ConfigSupplier<EventJournalConfig> configSupplier = new ConfigSupplier<EventJournalConfig>() {
```

Comment:
In addition to instantiating the strategy, here we instantiate an anonymous inner class each time we need the event journal config.